doc/howto.md

```diff
@@ -0,0 +1,428 @@
+# How to develop a new IOTAgent
+
+## Index
+
+* [Overview](#overview)
+* [Basic IOTA](#basic)
+* [IOTA With active attributes](#active)
+* [IOTA With lazy attributes](#lazy)
+* [IOTA With commands](#commands)
+* [Configuration control](#configuration)
+
+## <a name="overview"/> Overview
+This document's goal is to show how to develop a new IOT Agent step by step. To do so, a simple invented HTTP protocol
+will be used, so it can be tested with simple command line instructions as `curl` and `nc`. 
+
+### Protocol
+The invented protocol will be freely adapted from Ultralight 2.0. Whenever a device wants to send an update, it will
+send a request as the following:
+```
+curl -X GET 'http://127.0.0.1:8080/iot/d?i=ULSensor&k=abc&d=t|15,l|19.6' -i
+```
+Where:
+* **i**: is the device ID.
+* **k**: the API Key for the device's service.
+* **d**: the data payload, consisting of key/value pairs separated by a pipe ('|'), with each pair separated by comma (',');
+
+### Requirements
+
+This tutorial expects a Node.js v0.10 (at least) installed and working on your machine. It also expects you to have
+access to a Context Broker (without any security proxies).
+
+## <a name="basic"/> Basic IOTA
+In this first chapter, we will develop an IOT Agent with a fully working northbound API and no southbound
+API. This may seem useless (and indeed it is) but will serve us weel on showing the basic steps in the creation
+of a IOTA.
+ 
+First of all, we have to create the Node project. Create a folder to hold your project and type the following 
+instruction:
+```
+npm init
+```
+This will create the `package.json` file for our project. Now, add the following lines to your project file:
+```
```

Comment:
Not a good idea in that particular fragment, as it is really just a fragment. But I will add it to the Javascript fragments.